src/Teapot.Web/Controllers/StatusController.cs

```diff
@@ -1,86 +1,55 @@
-
-        public ActionResult Cors(int statusCode, int? sleep = SLEEP_MIN)
-        {
-            if (Request.HttpMethod != "OPTIONS")
-            {
-                return StatusCode(statusCode, null, sleep);
-            }
-
-            var allowedOrigin = Request.Headers.Get("Origin") ?? "*";
-            var allowedMethod = Request.Headers.Get("Access-Control-Request-Method") ?? "GET";
-            var allowedHeaders = Request.Headers.Get("Access-Control-Request-Headers") ?? "X-Anything";
-
-            var responseHeaders = new Dictionary<string, string>
-            {
-                { "Access-Control-Allow-Origin", allowedOrigin },
-                { "Access-Control-Allow-Headers", allowedHeaders },
-                { "Access-Control-Allow-Methods", allowedMethod }
-            };
-
-            var statusData = new StatusCodeResult { IncludeHeaders = responseHeaders };
-
-            DoSleep(sleep);
-
-            return new CustomHttpStatusCodeResult((int)HttpStatusCode.OK, statusData);
-        }
-
-        private static void DoSleep(int? sleep)
-        {
-            int sleepData = SanitizeSleepParameter(sleep, SLEEP_MIN, SLEEP_MAX);
-
-            if (sleepData > 0)
-            {
-                System.Threading.Thread.Sleep(sleepData);
-            }
-        }
-
-        private static int SanitizeSleepParameter(int? sleep, int min, int max)
-        {
-            var sleepData = sleep ?? 0;
-
-            // range check - minimum should be 0
-            if (sleepData < min)
-            {
-                sleepData = min;
-            }
-
-            // range check- maximum should be 300000 (5 mins)
-            if (sleepData > max)
-            {
-                sleepData = max;
-            }
-
-            return sleepData;
-        }
-    }
-}
+﻿using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Cors;
+using Microsoft.AspNetCore.Mvc;
+using Teapot.Web;
+using Teapot.Web.Models;
+
+namespace Teapot.Web.Controllers
+{
+    //[EnableCors]
+    public class StatusController : Controller
+    {
+        private static readonly TeapotStatusCodeResults StatusCodes = new TeapotStatusCodeResults();
+
+        [HttpGet("")]
+        public IActionResult Index() => View(StatusCodes);
+
+        [Route("{statusCode}", Name = "StatusCode")]
+        public async Task<IActionResult> StatusCode(int statusCode, int? sleep)
+        {
+            var statusData = StatusCodes.ContainsKey(statusCode)
+                ? StatusCodes[statusCode]
+                : new TeapotStatusCodeResult { Description = $"{statusCode} Unknown Code" };
+
+            await DoSleep(sleep);
+
+            return new CustomHttpStatusCodeResult(statusCode, statusData);
+        }
+
+        [EnableCors(Constants.CorsPolicy)]
+        [Route("{statusCode}/cors", Name = "Cors")]
+        public async Task<IActionResult> Cors(int statusCode, int? sleep)
+        {
+            var statusData = StatusCodes.ContainsKey(statusCode)
```

Comment:
As mentioned in the description, this now fully relies on the CORS middleware to handle preflight and CORS requests.

Reply:
Yeah, I feel like this can be dropped, since it simplifies it a lot using the middleware, unless you can configure CORS to only apply on a sub-path using the middleware

Reply:
The `[EnableCors]` attribute will only enable CORS for that action/endpoint, see <https://docs.microsoft.com/aspnet/core/security/cors?view=aspnetcore-3.1#enable-cors>

Reply:
Which I don't think we want, I think we want CORS to "just work".

It's worth remembering that CORS wasn't really _a thing_ when this was first written, and it was added in later.